pkg/provisioner/ironic/ironic.go

```diff
@@ -60,6 +60,19 @@ var bootModeCapabilities = map[metal3v1alpha1.BootMode]string{
 	metal3v1alpha1.Legacy: "boot_mode:bios",
 }
 
+type macAddressConflictError struct {
+	Address      string
+	ExistingNode string
+}
+
+func (e *macAddressConflictError) Error() string {
```

Comment:
nit: I don't think this has to be a pointer receiver (since you're not modifying it); removing the * would allow you to ditch all of the other * and & stuff (and the NewMacAddressConflictError function, whose only real purpose is adding the &).